### ISSUE TITLE
Compiler hangs in custom project

### ISSUE THREAD

**Issue by jaschdoc** [flix/flix#2434]
I've encountered an issue where the compiler hangs and takes up all system resources available. Normally, this is not an issue with the VSCode extension since it simply times out, however, when trying to compile and run the Flix project / program manually it problem occurs.
[This is the project I'm trying to run](https://github.com/jaschdoc/flix-develop).

**Comment by magnus-madsen** [comment]
Does this still happen? And what happens specifically?

**Comment by jaschdoc** [comment]
I think we should just close the issue as I apparently didn't post any technical details or commit hashes that could be referred to. I looked into [commit 5ef3584](https://github.com/jaschdoc/flix-develop/tree/5ef35846b9e9f322191ad8ec44f6628178d8ef96) which was the last commit made on the same date I posted this, but there have been lots of syntactic changes since so it would be a waste of time to look into this further.

If memory serves me right it was around the same time we had a parser bug on nested tags which resulted in extremely long compilation times so my best guess is that this issue was related to that (i.e. the parser would hog up a lot of memory). This was a big issue in the `DelayList` tests where I had used lots of nested tags.
